### ISSUE TITLE
[locale-modal]: modal trigger button does not announce country/language for screen readers

### ISSUE THREAD

**Issue by kennylam** [carbon-design-system/carbon-for-ibm-dotcom#8859]
### Description

The `locale-modal` trigger button does not announce the country/language for screen readers.

<img width="434" alt="image" src="https://user-images.githubusercontent.com/909118/168807675-e2231d73-6413-4b8e-8c43-9e695c8ca2e0.png">

### Component(s) impacted

`footer`

### Browser

Chrome

### Carbon for IBM.com version

v1.34.0

### Severity

Severity 3 = The problem is visible or noticeable to users but does not impede the usability or functionality. Affects minor functionality, has a workaround.

### Application/website

Carbon for IBM.com

### Package

@carbon/ibmdotcom-web-components

### CodeSandbox example

https://www.ibm.com/standards/carbon/web-components/?path=/story/components-footer--default

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues) for duplicate issues
